### ISSUE TITLE
Some instances of VSTHRD010 do not report the location of the error.

### ISSUE THREAD

**Issue by lgolding** [Microsoft/vs-threading#314]
#### Bug description
Some instances of VSTHRD010 do not report the location of the error. Here's an example:
```
CSC: error VSTHRD010: Add call to Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread() at start of member body bec
ause this member invokes other members that require the main thread. [c:\Code\sarif-sdk\src\Sarif.Viewer.VisualStudio\Sarif.Vie
wer.VisualStudio_zyfzy4vp_wpftmp.csproj]
```
Note that the message begins with "CSC:" and does not tell me the file name or line number.
#### Repro steps
Steps or code to reproduce the behavior:
1. `git clone https://github.com/Microsoft/sarif-sdk`
2. cd to the root of the repo.
3. `git checkout -b origin/users/lgolding/vsthrd010`
4. Run the command: `BuildAndTest.cmd /notest /nopublish`

#### Expected behavior
There should be a single instance of VSTHRD010 (because I uncommented one of my calls to `ThrowIfNotOnUIThread`):
```
CodeAnalysisResultManager.cs(97,13): error VSTHRD010: Add call to Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThrea
d() at start of member body because this member invokes other members that require the main thread. [c:\Code\sarif-sdk\src\Sari
f.Viewer.VisualStudio\Sarif.Viewer.VisualStudio_etyp0syf_wpftmp.csproj]
```
Note that the message begins with the file name and line number.

#### Actual behavior
There are five instances of VSTHRD010, and four of them don't tell me the location.

- Version used: Microsoft.VisualStudio.Threading.Analyzers.15.7.17

#### Additional context
None

@michaelcfanning FYI

**Comment by sharwell** [comment]
Duplicate of #246

**Comment by AArnott** [comment]
Agreed. This is fixed in the 15.8 release of the analyzers.
Thanks, @sharwell, and @lgolding for your report.

**Comment by lgolding** [comment]
Thanks all, and thanks for the pointer to the version with the fix.
